### ISSUE TITLE
Activity Log not updating for metadata changes

### ISSUE THREAD

**Issue by salmanm2003** [pkp/pkp-lib#10019]
Hi @asmecher,
Updating contributor metadata in OJS is only logged in the Activity Log if it pertains to the primary author. Changes to co-authors' details are not reflected in the Activity Log, even though they are updated in the UI. This omission results in hidden actions during audits and creates issues during the production process.

**To Reproduce**
1- go to publication > Contributors
2- edit/add a new contributor
3- check the Activity Log and there wont be any new logs
4- edit the primary author data and the log is going to be there 

**What application are you using?**
 OJS version 3-3-0_17

**Additional information**
[https://forum.pkp.sfu.ca/t/activity-log-not-updating-for-metadata-changes/88324](https://forum.pkp.sfu.ca/t/activity-log-not-updating-for-metadata-changes/88324)

**Solution**
I could fix it by editing this file:
[https://github.com/pkp/pkp-lib/blob/6e395415a13f0c4e155218ca5cfcd0f6a7e11fef/controllers/grid/users/author/form/PKPAuthorForm.inc.php#L247](https://github.com/pkp/pkp-lib/blob/6e395415a13f0c4e155218ca5cfcd0f6a7e11fef/controllers/grid/users/author/form/PKPAuthorForm.inc.php#L247)
by adding:
[https://github.com/ubiquitypress/pkp-lib/blob/ff6d4ccbef4bfb3e3d9c490af2289453fe434feb/controllers/grid/users/author/form/PKPAuthorForm.inc.php#L247-L265](https://github.com/ubiquitypress/pkp-lib/blob/ff6d4ccbef4bfb3e3d9c490af2289453fe434feb/controllers/grid/users/author/form/PKPAuthorForm.inc.php#L247-L265)

If this properly fixes the issue, please let me know so I can send a PR.

PRs:
stable-3_3_0:
- https://github.com/pkp/pkp-lib/pull/10087
- https://github.com/pkp/ojs/pull/4328 (only submodule update)
- https://github.com/pkp/omp/pull/1600 (only submodule update)
- https://github.com/pkp/ops/pull/711 (only submodule update)

stable-3_4_0:
- https://github.com/pkp/pkp-lib/pull/10091
- https://github.com/pkp/ojs/pull/4329 (only submodule update)
- https://github.com/pkp/omp/pull/1601 (only submodule update)
- https://github.com/pkp/ops/pull/712 (only submodule update)

main:
- https://github.com/pkp/pkp-lib/pull/10092

**Comment by asmecher** [comment]
@bozana, could you look at this one? Thanks!

**Comment by bozana** [comment]
Hi @salmanm2003,
The editing of the primary contact by chance adds the entry in the activity log -- just because the attribute 'primaryContactId' is actually a publication attribute/parameter. For some reason we have not considered adding/editing/deleting authors for the activity log, and this can be changed, I think.
Long story short :-) : 
Your changes are correct, and will also consider adding an author. But deleting an author is not considered. Would it be possible for you to also add this code:
```
// Log an event when publication data is updated
$submissionId = (int) $request->getUserVar('submissionId');
$submission = Services::get('submission')->get($submissionId);
import('lib.pkp.classes.log.SubmissionLog');
import('classes.log.SubmissionEventLogEntry');
SubmissionLog::logEvent(Application::get()->getRequest(), $submission, SUBMISSION_LOG_METADATA_UPDATE, 'submission.event.general.metadataUpdated');
```
here: https://github.com/pkp/pkp-lib/blob/stable-3_3_0/controllers/grid/users/author/AuthorGridHandler.inc.php#L394 
when you make the PR? Else, I can add it...

Thanks a lot!

**Comment by salmanm2003** [comment]
@bozana Thanks for the clarification. I can add it to deleting authors as well.

Additionally, there is one more thing that we can do that might help more. When adding, editing, or deleting contributor data, we can call `publication->edit($params)` where `$params = []`. This will notify us that the publication has been updated. 

I understand this is more of a workaround than a bug fix, but it will address the logging issue and allow the plugins to recognize that the publication has been updated?

**Comment by bozana** [comment]
Hi @salmanm2003, yes, that sound good! This will also change `last_modified` date for that publication.
Thanks a lot!

p.s.
And I will double check in our dev channel if we need to get those two submission objects in the function edit() -- it seems the first submission can be moved into the `if (array_key_exists('coverImage', $params)) {` and the second is maybe not needed... :thinking:

**Comment by salmanm2003** [comment]
Hi @bozana,
here is the PR: [https://github.com/pkp/pkp-lib/pull/10087/commits](#10087)

**Comment by bozana** [comment]
Hi @salmanm2003, I provided the sumodule updates and ported the changes to stable-3_4_0 and main. Everything is merged now.
Thanks a lot for the contribution!

**Comment by fradeve** [comment]
@bozana do you have any feeling of when 3.3.0-18 will be released? If I understand correctly, the work on this issue will be included in 3.3.0-18?
Thanks

**Comment by bozana** [comment]
Hi @fradeve, we are planning the next release(s) for middle of July. 
Yes, this should be included in the next release 3.3.0-18 (and 3.4.0.6).
